cloud-aws-common/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/common/AwsObjectStorageConnector.java

```diff
@@ -104,6 +115,23 @@ public ObjectStorageValidateResponse validateObjectStorage(ObjectStorageValidate
         return response;
     }
 
+    @Override
+    public ObjectStorageSizeResponse getObjectStorageSize(ObjectStorageSizeRequest request) {
+        GetMetricStatisticsResult result = cloudWatchCommonService.getBucketSize(request.getCredential(), request.getRegion().value(),
+                request.getStartTime(), request.getEndTime(), request.getObjectStoragePath());
+        Optional<Datapoint> latestDatapoint = result.getDatapoints().stream().max(Comparator.comparing(Datapoint::getTimestamp));
+        if (latestDatapoint.isPresent()) {
+            Datapoint datapoint = latestDatapoint.get();
+            LOGGER.debug("Gathered datapoint from CloudWatch: {}", datapoint);
+            return ObjectStorageSizeResponse.builder().withStorageInBytes(datapoint.getMaximum()).build();
+        } else {
```

Comment:
else is unnecessary, because you return in the if branch

Reply:
I prefer to use else branches in these situations because imo it's easier determine when looking at the function that what are the possible outcomes.

Reply:
it ok for me, just a hint